src/qtrendplot.cpp

```diff
@@ -78,7 +89,25 @@ void QTrendPlot::removeSelectedGraphs(){
 }
 
 void QTrendPlot::addSelectedGraphs(){
-    this->addGraphList(casuTree->selectedItems());
+    QList<QTreeWidgetItem*> selectedList = casuTree->selectedItems();
+
+    foreach(QTreeWidgetItem* item, groupTree->selectedItems()){
+        QTreeWidgetItem* parent = item->parent();
+        while(parent->parent()) parent = parent->parent();
+        QColor color = parent->textColor(0);
+        QString name = item->text(0);
+        QString parentName = parent->text(0);
+
+        foreach (QGraphicsItem* casuItem, arenaScene->items()) {
+            if(casuItem->childItems().size()) continue;
+            if(((QCasuSceneItem*)casuItem)->groupColor != color && !QString::compare(parentName, "CASU group")) continue;
+            if(!casuItem->isSelected() && !QString::compare(parentName, "Selected CASUs")) continue;
+            selectedList.append(((QCasuSceneItem*)casuItem)->treeItem->widgetMap[name]);
+        }
+    }
```

Comment:
Code for finding all dataBuffers from itemGroup members is very dirty. QCasuTreeItem doesn't know his QCasuSceneItem counterpart. When plotting group and its members, all have assigned groupColor which the same as their text color in groupTree. We use that information to find corresponding QCasuSceneItems and their QCasuTreeItems (scene item knows which tree item is theirs).

After finding correct QCasuTreeItem we search its children (maped by unique dataName) and add them to selectedList which is at the end added to trend.